### DISCUSSION TITLE
Does the flag `-cert-secret-ref` is still functional while creating a ImageRepository source. ?

### DISCUSSION THREAD

**Discussion by karnatisrinivas** [fluxcd/flux2#4201] [General]
Does the flag `-cert-secret-ref` is still functional while creating a ImageRepository source. I was working with Flux to create a ImageRepository as source which is an private registry. I have followed docs and created tls secret with cert file and key file. Documentation contains to create ImageRepo source as follows : 
```
flux create image repository app-repo \
    --cert-secret-ref client-cert \
    --image registry.example.com/private/app --interval 5m
```
I have replicated the same and this is the reponse I got. 

![image](https://github.com/fluxcd/flux2/assets/52213014/be6f9b0c-821c-4e02-98a0-d14293e94466)

Also the help command also does not show this particular flag 

```
flux create image repository --help | grep --cert-secret-ref
grep: unrecognized option '--cert-secret-ref'
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.
````

**Comment by makkes** [comment]
The flag is called `--cert-ref`.

I suppose the documentation you're referring to is [this one](https://fluxcd.io/flux/cmd/flux_create_image_repository/). This is indeed wrong and needs to be fixed.

**Reply by karnatisrinivas** [reply]
If that is an issue, can I open a issue? Also, I am interested in working on this one. 😍

**Reply by makkes** [reply]
Yes, please create an issue in this repository. I'd be happy to assign it to you.

**Reply by karnatisrinivas** [reply]
https://github.com/fluxcd/flux2/issues/4202
